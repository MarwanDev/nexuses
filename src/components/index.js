export { default as Navbar } from './Navbar/Navbar';
export { default as Layout } from './Layout/Layout';
export { default as Header } from './Header/Header';
export { default as Footer } from './Footer/Footer';
export { default as DoctorSlider } from './DoctorSlider/DoctorSlider';
export { default as DoctorCard } from './DoctorCard/DoctorCard';
export { default as Table } from './Table/Table';

