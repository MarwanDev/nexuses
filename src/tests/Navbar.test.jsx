import { Navbar } from "../components";
import { BrowserRouter } from "react-router-dom";
import { describe, it, expect } from "vitest";
import renderer from "react-test-renderer";

describe("Navbar", () => {
  it("renders navbar correctly", () => {
    const navbar = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      )
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });
});
