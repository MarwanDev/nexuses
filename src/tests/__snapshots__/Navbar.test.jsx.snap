// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > renders navbar correctly 1`] = `
<nav
  className="navbar-container"
>
  <img
    alt="logo"
    src="/src/assets/Nexuses_logo_0_002.webp"
    style={
      {
        "height": 38,
        "width": 100,
      }
    }
  />
  <ul
    className="links-container"
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "fontSize": 20,
        "fontWeight": 700,
        "justifyContent": "space-between",
        "listStyleType": "none",
        "textDecoration": "none",
        "width": "40%",
      }
    }
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
        style={
          {
            "color": "#3684B0",
            "textDecoration": "none",
          }
        }
      >
        Explore
      </a>
    </li>
    <li>
      <a
        href="/doctors"
        onClick={[Function]}
        style={
          {
            "color": "#3684B0",
            "textDecoration": "none",
          }
        }
      >
        My Doctors
      </a>
    </li>
    <li>
      <a
        href="/med-history"
        onClick={[Function]}
        style={
          {
            "color": "#3684B0",
            "textDecoration": "none",
          }
        }
      >
        My Medical History
      </a>
    </li>
  </ul>
  <div
    className="navbar-menu"
  >
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": "#3684B0",
          "fontSize": 25,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
      />
    </svg>
  </div>
</nav>
`;
